service-loadbalancer/service_loadbalancer.go

```diff
@@ -124,6 +124,10 @@ type service struct {
 	// for this service. For http, it's always :80, for each tcp service it
 	// is the service port of any service matching a name in the tcpServices set.
 	FrontendPort int
+
+	// Host if not empty it will add a new haproxy acl to route traffic using the
```

Comment:
> I mean if I have a single requests that matches url and host header, is my backend invoked twice?

I changed the template to simplify the matching. It uses multiple acls and one `use_backend` directive with an `or` condition. The order is `url_acl` and then `host_acl`.